dwitter/templates/base.html

```diff
@@ -36,32 +36,38 @@
   </head>
   <body class="{% block body_class %}{% endblock %}">
     <script type="text/javascript">
-      // Check for compatibility
-      if (window.localStorage) {
-        var switchTheme = function() {
-          // Shift themes
-          var theme = localStorage.getItem('theme');
-          if (!theme) {
-            localStorage.setItem('theme', 'dark');
-          } else if (theme === 'dark') {
-            localStorage.setItem('theme', 'green');
-          } else if (theme === 'green') {
-            localStorage.setItem('theme', 'hcontrast');
-          } else {
-            localStorage.setItem('theme', '');
-          }
-          location.reload();
-          // Call this function in the li to change the mode
+      (function() {
+        var themes = ['', 'dark', 'green', 'hcontrast'];
+        var themeUrls = {
+          dark: '{% static "theme_dark.css" %}',
+          green: '{% static "theme_green.css" %}',
+          hcontrast: '{% static "theme_hcontrast.css" %}'
         };
+        // Use saved theme if localStorage is available
+        var theme = window.localStorage ? localStorage.getItem('theme') : null;
+        // Ignore unknown theme
+        if (themes.indexOf(theme) == -1) {
+          theme = null;
+        }
+        // Use dark mode preference if no theme is set
+        if (typeof theme != 'string' && window.matchMedia) {
+          theme = matchMedia('(prefers-color-scheme: dark)').matches ? themes[1] : themes[0];
+        }
+
+        // Check for compatibility
+        if (window.localStorage) {
+          window.switchTheme = function() {
+            // Shift themes
+            theme = themes[(themes.indexOf(theme) + 1) % themes.length];
+            localStorage.setItem('theme', theme);
+            location.reload();
```

Comment:
While I’m at it, is there some compatibility issue with loading the CSS files dynamically without a reload? It wouldn’t be a big change to get rid of the reload otherwise.

Reply:
It seems to me that the reload is just a quick way to re-run the code that sets the theme (which only runs once on boot). But that could easily be wrapped in its on function that is both run on boot and re-run when switching themes, I'd think.

I'll have a quick look in the change history of the file to see if there's any other reason I couldn't think of!

Reply:
Yup, looks like it was just written like that from the start - I think that should be a quick change and just work :+1:

Reply:
Added it. Checked locally in FF, Chrome, and Safari.